controllers/storagecluster/volumegroupsnapshotterclasses.go

```diff
@@ -0,0 +1,169 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	groupsnapapi "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumegroupsnapshot/v1alpha1"
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumesnapshot/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type GroupSnapshotterType string
+
+type ocsGroupSnapshotClass struct{}
+
+const (
+	rbdGroupSnapshotter    GroupSnapshotterType = "rbd"
+	cephfsGroupSnapshotter GroupSnapshotterType = "cephfs"
+)
+
+const (
+	groupSnapshotterSecretName      = "csi.storage.k8s.io/group-snapshotter-secret-name"
+	groupSnapshotterSecretNamespace = "csi.storage.k8s.io/group-snapshotter-secret-namespace"
+)
+
+type GroupSnapshotClassConfiguration struct {
+	groupSnapshotClass *groupsnapapi.VolumeGroupSnapshotClass
+	reconcileStrategy  ReconcileStrategy
+	disable            bool
+}
+
+var driverName, driverValue string
+
+func newVolumeGroupSnapshotClass(instance *ocsv1.StorageCluster, groupSnaphotterType GroupSnapshotterType) *groupsnapapi.VolumeGroupSnapshotClass {
+	driverName, driverValue = setParameterBasedOnSnapshotterType(instance, groupSnaphotterType)
+	groupSnapClass := &groupsnapapi.VolumeGroupSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generateNameForGroupSnapshotClass(instance, groupSnaphotterType),
+		},
+		Driver: generateNameForSnapshotClassDriver(SnapshotterType(groupSnaphotterType)),
+		Parameters: map[string]string{
+			"clusterID":                     instance.Namespace,
```

Comment:
where are we setting filesystem name and blockpool in the class?

Reply:
it is set by the `driverName` and `driverValue` variables whose values are assigned by the `setParameterBasedOnSnapshotterType` function.

Reply:
Right, this confused me a bit too, it is well hidden. `driverName` is not really a very suitable name, as the parameter is called `fsName` for CephFS or `pool` for RBD. But, I expect this to work correctly. Having a test that validates these kind of parameters would be nice to have.